lib/http.c

```diff
@@ -1919,22 +1918,28 @@ CURLcode Curl_http(struct connectdata *conn, bool *done)
 #ifdef HAVE_LIBZ
   /* we only consider transfer-encoding magic if libz support is built-in */
 
-  if(!Curl_checkheaders(conn, "TE:") &&
+  if(!Curl_checkheaders(conn, "TE") &&
      data->set.http_transfer_encoding) {
     /* When we are to insert a TE: header in the request, we must also insert
        TE in a Connection: header, so we need to merge the custom provided
        Connection: header and prevent the original to get sent. Note that if
        the user has inserted his/hers own TE: header we don't do this magic
        but then assume that the user will handle it all! */
-    char *cptr = Curl_checkheaders(conn, "Connection:");
+    char *cptr = Curl_checkheaders(conn, "Connection");
 #define TE_HEADER "TE: gzip\r\n"
 
     Curl_safefree(conn->allocptr.te);
 
+    cptr = Curl_copy_header_value(cptr);
+    if(!cptr)
+      return CURLE_OUT_OF_MEMORY;
+
     /* Create the (updated) Connection: header */
-    conn->allocptr.te = cptr? aprintf("%s, TE\r\n" TE_HEADER, cptr):
+    conn->allocptr.te = (cptr && *cptr) ?
+      aprintf("%s, TE\r\n" TE_HEADER, cptr) :
```

Comment:
Now ``cptr`` contains only the header value.

``"%s, TE\r\n"`` must be changed to ``"Connection: %s, TE\r\n"``

Reply:
ack, and it even broke some test cases because of this...